src/mock-doc/parser.ts

```diff
@@ -0,0 +1,16 @@
+import { MockDocument } from './document';
+import { parseHtmlToDocument } from './parse-html';
+
+export type DOMParserSupportedType =
+  | 'text/html'
+  | 'text/xml'
+  | 'application/xml'
+  | 'application/xhtml+xml'
+  | 'image/svg+xml';
+
+export class MockDOMParser {
+  parseFromString(string: string, type: DOMParserSupportedType): MockDocument {
+    if (type !== 'text/html') console.error('XML parsing not implemented yet, continuing as html');
```

Comment:
Nitpick: Can we add curly braces around the 'then' part of this conditional?

```suggestion
    if (type !== 'text/html') {
      console.error('XML parsing not implemented yet, continuing as html');
    }
```